### PR TITLE
Continuous decompression

### PR DESCRIPTION
**Pull request by Fonsan**
Added option for continuous  decompression of joined compressed buffers

### CODE REVIEW

File: ext/zstdruby/zstdruby.c [hunk 1]
```diff
@@ -79,15 +79,23 @@ static VALUE decompress_buffered(const char* input_data, size_t input_size)
   return output_string;
 }
 
-static VALUE decompress(VALUE self, VALUE input)
+static VALUE decompress(int argc, VALUE *argv, VALUE self)
 {
+  VALUE input = argv[0];
```
Comment (SpringMT) [inline ext/zstdruby/zstdruby.c]: `rb_scan_args` is better for parsing args.
How about the below way?

```
VALUE input;
VALUE continuous_decompression_value;
rb_scan_args(argc, argv, "11", & input, &continuous_decompression_value);

VALUE continuous_decompression;
if (NIL_P(continuous_decompression_value)) {
    continuous_decompression = RUBY_T_FALSE;
}
```

File: spec/zstd-ruby_spec.rb [hunk 2]
```diff
@@ -33,6 +33,12 @@
       expect(decompressed).to eq('abc')
     end
 
+    it 'should work with two continous compressed buffers' do
```
Comment (SpringMT) [inline spec/zstd-ruby_spec.rb]: 👍

File: ext/zstdruby/zstdruby.c (hunk 1 above)
Comment (Watson1978) [reply ext/zstdruby/zstdruby.c:84]: `RUBY_T_FALSE` is class type. I think you should use `Qtrue` or `Qfalse` instead in this case.

**Comment by SpringMT** [comment]
Support Streaming Decompression https://github.com/SpringMT/zstd-ruby/pull/43.

**Pull request closed without merging**